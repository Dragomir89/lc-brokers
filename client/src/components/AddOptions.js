import React, { Component } from 'react';
import TextInput from './common/TextInput';

class AddOpitonsPage extends Component {

    state = {
        constructionType: '',
        estateType: '',
        state: '',
        neighborhood: ''
    }

    changeFn = (e) => {
        const { name, value } = e.target;
        this.setState(() => {
            return { [name]: value }
        });
    }

    handleClick = (e) => {
        e.preventDefault();
        console.log(this.state);
        this.setState(() => {
            return  {
                constructionType: '',
                estateType: '',
                state: '',
                neighborhood: ''
            } 
        });
    }

    render () {
        return (
            <div className='content-container'>
                <h1 className='title__page'>Добвавете опции за офертите</h1>
                <form>
                    <TextInput
                        value={this.state.constructionType}
                        label='Вид Стоителство'
                        name='constructionType'
                        placeholder='старо, ново, ЕПК ...' changeFn={this.changeFn} />
                    <TextInput
                        value={this.state.estateType}
                        label='Вид Имот'
                        name='estateType'
                        placeholder='едностаен, двустаен, къща ...'
                        changeFn={this.changeFn} />
                    <TextInput
                        value={this.state.state}
                        label='Състояние'
                        name='state'
                        placeholder='нает, свободен ...'
                        changeFn={this.changeFn} />
                    <TextInput
                        value={this.state.neighborhood}
                        label='Квартал'
                        name='neighborhood'
                        placeholder='лозенец, цетър, стрелбище ...'
                        changeFn={this.changeFn} />
                    <button 
                        onClick={this.handleClick}
                        className='button__primary'>
                        Запази
                    </button>
                </form>
            </div>
        )
    }
}

export default AddOpitonsPage;