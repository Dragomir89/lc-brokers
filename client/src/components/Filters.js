import React from 'react';
import { connect } from 'react-redux';
import moment from 'moment';
import qs from 'querystring';
import TextInput from './common/TextInput';
import Select from './common/Select';
import { getOptions } from '../actions/optionsActions';
import { getOffers } from '../actions/offerActions';
import CustomSingleDatePicker from './common/SingleDatePicher';

class Filters extends React.Component {
    state = {
        focused: false,
        phoneNumber: '',
        nextCall: moment()
    }


    componentDidMount () {
        this.props.getOptions();
    }

    onChahgeHandler = (e) => {
        const { name, value } = e.target;
        this.setState(() => ({ [name]: value }));
    }

    handleClick = (e) => {
        e.preventDefault();
        const querystring = qs.stringify(
            { ...this.state, nextCall: this.state.nextCall.format("YYYY-MM-DD") });
        this.props.getOffers(1, querystring);
    }

    onDateChange = (nextCall) => { this.setState(() => ({ nextCall })) };

    onFocusChange = ({ focused }) => { this.setState(() => ({ focused: focused })) }

    render () {
        const style01 = {
            display: 'flex',
            justifyContent: 'space-between'
        }
        const style02 = {
            width: '25%'
        }
        const { constructionTypes, propertyTypes, states, neighborhoods } = this.props.options;
        return (
            <form>
                <div style={style01}>
                    <div style={style02}>
                        <Select
                            options={constructionTypes}
                            label='Строителство'
                            name='constructionTypeId'
                            changeFn={this.onChahgeHandler} />
                    </div>
                    <div style={style02}>
                        <Select

                            options={propertyTypes}
                            label='Вид Имот'
                            name='propertyTypeId'
                            changeFn={this.onChahgeHandler} />
                    </div>
                    <div style={style02}>
                        <Select
                            options={states}
                            label='Състояние'
                            name='state'
                            placeholder='Телефон'
                            changeFn={this.onChahgeHandler} />
                    </div>

                </div>
                <div style={style01}>
                    <div style={style02}>
                        <Select
                            options={neighborhoods}
                            label='Квартал'
                            name='neighborhoodId'
                            changeFn={this.onChahgeHandler} />
                    </div>
                    <div style={style02}>
                        <CustomSingleDatePicker
                            label='Следващо обаждане'
                            date={this.state.nextCall}
                            onDateChange={this.onDateChange}
                            focused={this.state.focused}
                            onFocusChange={this.onFocusChange}
                            numberOfMonths={1}
                        />
                        
                    </div>
                    <div style={style02}>
                        <TextInput
                            value={this.state.phoneNumber}
                            label='Телефон'
                            name='phoneNumber'
                            placeholder='Телефон'
                            changeFn={this.onChahgeHandler} />
                    </div>

                </div>
                <button
                    style={{ width: '100%' }}
                    className='button__primary'
                    onClick={this.handleClick}>
                    Търси
                </button>
            </form>
        )
    }
}

const mapStateToProps = (state) => ({ options: state.options });
const actions = { getOptions, getOffers }
export default connect(mapStateToProps, actions)(Filters);