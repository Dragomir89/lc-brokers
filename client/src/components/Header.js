import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => {
    return(
        <div className='header'>
            <div  className='header-layout content-container '>
                <Link className='header-link' to='/'>LC Brokers</Link>
                
                <Link className='header-link' to='/add-options'>Добави опции</Link>
                
                <Link className='header-link' to='/offers'>Оферти</Link>

                <Link className='header-link' to='/add-offer'>Добави Оферта</Link>   
            </div>
        </div>
    )
}

export default Header;