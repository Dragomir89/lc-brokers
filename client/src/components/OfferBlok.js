import React from 'react';
import { Link } from 'react-router-dom';

const OfferBlock = ({ address, price, area, floor, neighborhood, estateType, _id }) => {

    return (
        <div className='content-container__offer-wrapper'>
            <div >
                <img height='180px' src='https://t-ec.bstatic.com/images/hotel/max1024x768/137/137238081.jpg' />
            </div>
            <div style={
                {
                    marginLeft: '5px',
                    height: '180px',
                    position: 'relative'
                }

            }>
                <div className='test'>
                    <h3 className='title__offer'> {neighborhood} </h3>
                    <h3 className='title__offer'> {estateType} </h3>

                </div>

                <div className='offer-info-line'>
                    Адрес: {address} - етаж: {floor}
                </div>

                <div className='offer-info-line'>
                    Площ: {area}кв.
                    </div>
                <div>
                    Цена: {price} евро
                    </div>
                <div className='button__offer-details-position'>
                    <Link to={`/edit-offer/${_id}`}>
                        <button className='button__offer-details'>
                            Виж Детайли
                        </button>
                    </Link>
                </div>
            </div>
        </div>
    )
}

export default OfferBlock;
