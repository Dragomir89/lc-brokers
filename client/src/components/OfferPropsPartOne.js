import React from 'react';

import Select from './common/Select';
import TextInput from './common/TextInput';
import { connect } from 'react-redux';
import { getOptions } from '../actions/optionsActions';

class OfferPropsPartOne extends React.Component {

    componentDidMount () {
        this.props.getOptions()
    }

    render () {
        const { constructionTypes, propertyTypes, states, neighborhoods } = this.props.options;
        const { constructionTypeId, propertyTypeId, state, neighborhoodId } = this.props.values;
        const { floor, description, area, price } = this.props.values;
        return (
            <div>
                <div className='form__add-offer-layout'>
                    <div className='form__add-offer-section'>
                        <Select
                            selectedOptionId={constructionTypeId}
                            options={constructionTypes}
                            label='Вид Стоителство'
                            name='constructionTypeId' changeFn={this.props.changeHandler} />
                        <Select
                            selectedOptionId={propertyTypeId}
                            options={propertyTypes}
                            label='Вид Имот'
                            name='propertyTypeId' changeFn={this.props.changeHandler} />
                        <Select
                            selectedOptionId={state}
                            options={states}
                            label='Състояние'
                            name='state' changeFn={this.props.changeHandler} />
                        <Select
                            selectedOptionId={neighborhoodId}
                            options={neighborhoods}
                            label='Квартал'
                            name='neighborhoodId' changeFn={this.props.changeHandler} />
                    </div>
                    <div className='form__add-offer-section'>
                        <TextInput
                            label='Цена'
                            name='price'
                            value={price}
                            placeholder='Евро' changeFn={this.props.changeHandler} />
                        <TextInput
                            label='Квадратура'
                            name='area'
                            value={area}
                            placeholder='кв/м' changeFn={this.props.changeHandler} />
                        <TextInput
                            label='Описание'
                            name='description'
                            value={description}
                            placeholder='Описание' changeFn={this.props.changeHandler} />
                        <TextInput
                            label='Етаж'
                            name='floor'
                            value={floor}
                            placeholder='1, 2, 3 ...' changeFn={this.props.changeHandler} />
                    </div>
                </div>
                <div className='button__relative-wrapper'>
                    <button
                        onClick={(e) => {
                            e.preventDefault();
                            this.props.nextHandler();
                        }}
                        className='button__success button__offer-details-position'>Продължи</button>
                </div>
            </div>
        )
    }
}


const mapStateToProps = (state) => ({ options: state.options });
const actions = { getOptions }

export default connect(mapStateToProps, actions)(OfferPropsPartOne);