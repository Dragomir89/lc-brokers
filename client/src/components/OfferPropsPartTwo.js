import React from 'react';
import TextInput from './common/TextInput';
import TextArea from './common/TextArea';
import CustomSingleDatePicker from './common/SingleDatePicher';

export const OfferPropsPartTwo = (
    { changeHandler, backHandler, checkOfferHandler, values, prevCallChangeHandler,nextCallChangeHandler }) => {
    
    const { number, address, info, propertyOwnerName } = values;
    const { phoneNumber, phoneNumber2, phoneNumber3, lastCall, nextCall } = values;
    
    return (
        <div>
            <div className='form__add-offer-layout'>
                <div className='form__add-offer-section'>
                    <TextInput
                        label='Номер На Оферта'
                        name='number'
                        value={number}
                        placeholder='123456 ...' changeFn={changeHandler} />
                    <TextInput
                        label='Адрес'
                        name='address'
                        value={address}
                        placeholder='Улица, номер, етаж ...' changeFn={changeHandler} />
                    <TextArea
                        label='Долълнително Инфо'
                        name='info'
                        value={info}
                        placeholder='Инфо ...' changeFn={changeHandler} />
                    <CustomSingleDatePicker
                        label='Предишно обаждане'
                        date={lastCall}
                        onDateChange={prevCallChangeHandler}
                        numberOfMonths={1}
                    />
                </div>
                <div className='form__add-offer-section'>
                    <TextInput
                        label='Име на Собственик'
                        name='propertyOwnerName'
                        value={propertyOwnerName}
                        placeholder='Любомир Петров ...' changeFn={changeHandler} />
                    <TextInput
                        label='Тлефон Главен'
                        name='phoneNumber'
                        value={phoneNumber}
                        placeholder='889358720' changeFn={changeHandler} />
                    <TextInput
                        label='Тлефон 2'
                        name='phoneNumber2'
                        value={phoneNumber2}
                        placeholder='Не е задължителен' changeFn={changeHandler} />
                    <TextInput
                        label='Тлефон 3'
                        name='phoneNumber3'
                        value={phoneNumber3}
                        placeholder='Не е задължителен' changeFn={changeHandler} />
                    <CustomSingleDatePicker
                        label='Следващо обаждане'
                        date={nextCall}
                        onDateChange={nextCallChangeHandler}
                        numberOfMonths={1}
                    />
                </div>
            </div>
            <div className='button__relative-wrapper'>
                <button onClick={(e) => { e.preventDefault(); backHandler(); }}
                    className='button__success'>Назад</button>
                <button onClick={(e) => { e.preventDefault(); checkOfferHandler(); }}
                    className='button__primary button__offer-details-position'>Преглед на офертата</button>
            </div>
        </div>
    )
}

export default OfferPropsPartTwo;