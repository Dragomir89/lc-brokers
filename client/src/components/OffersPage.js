import React, { Component } from 'react';
import OfferBlok from './OfferBlok';
import { connect } from 'react-redux';
import { getOffers } from '../actions/offerActions';
import Filters from './Filters';

class OffersPage extends Component {

    render () {        
        const offers = this.props.offers.map(offer => (<OfferBlok key={offer._id} {...offer} />));
        return (
            <div className='content-container'>
                <Filters />
                {this.props.offers.length > 0 && 
                    (<div><h1 className='title__page'>Оферти</h1>{offers} </div>) }
            </div>
        )
    }
}

const mapStateToProps = (state) => ({ offers: state.offers });
const actions = { getOffers }

export default connect(mapStateToProps, actions)(OffersPage);