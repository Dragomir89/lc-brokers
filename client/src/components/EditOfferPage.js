import React, { Component } from 'react';
import { connect } from 'react-redux';
import OfferPropsPartOne from './OfferPropsPartOne';
import OfferPropsPartTwo from './OfferPropsPartTwo';
import moment from 'moment';
import { EDIT_POST_OFFER_SUCCESS, EDIT_POST_OFFER_ERROR } from '../actions/offerActionTypes';
import { postOffer, getCurrentOffer } from '../actions/offerActions';
import EditOfferModal from './modals/EditOfferModal';

class EditOfferPage extends Component {
    constructor(props) {
        super(props);
        this.defaultState = {
            showFierstPart: true,
            constructionTypeId: '',
            constructionType: '',
            propertyTypeId: '',
            propertyType: '',
            state: '',
            stateVal: '',
            neighborhoodId: '',
            neighborhood: '',
            price: '',
            area: '',
            description: '',
            floor: '',
            number: '',
            address: '',
            info: '',
            propertyOwnerName: '',
            phoneNumber: '',
            phoneNumber2: '',
            phoneNumber3: '',
            lastCall: moment(),
            nextCall: moment(),
            showAddOfferModal: false,
            offerPostStatus: ''
        }
        this.state = { ...this.defaultState };
    }

    componentDidMount () {
        const { id } = this.props.match.params;
        this.props.getCurrentOffer(id);
    }

    checkForCurrentOffer = (newProps) => {
        if (newProps.editOffer) {
            this.setState(() => ({ ...newProps.editOffer }));
        }
    }

    handleRequiestStatus = (newProps) => {
        if (newProps.offerPostStatus === EDIT_POST_OFFER_SUCCESS) { /// to be fixed
            this.setState(() => ({ showAddOfferModal: false }));
        } else if (newProps.offerPostStatus === EDIT_POST_OFFER_ERROR) {
            this.setState(() => ({ 
                offerPostStatus: EDIT_POST_OFFER_ERROR 
            }));
        }
    }

    componentWillReceiveProps (newProps) {
        this.checkForCurrentOffer(newProps);
        this.handleRequiestStatus(newProps);
    }

    selectedSetSelectedTextValues = (e, name) => {
        if (e.target.options) {
            const index = e.target.options.selectedIndex;
            const options = e.target.options;
            const value = options[index].innerText;

            if (name === 'constructionTypeId') {
                this.setState(() => ({ constructionType: value }));
            } else if (name === 'propertyTypeId') {
                this.setState(() => ({ propertyType: value }));
            } else if (name === 'state') {
                this.setState(() => ({ stateVal: value }));
            } else if (name === 'neighborhoodId') {
                this.setState(() => ({ neighborhood: value }));
            }
        }

    }

    changeHandler = (e) => {
        const { name, value } = e.target;
        this.selectedSetSelectedTextValues(e, name);
        this.setState(() => ({ [name]: value }));
    }

    prevCallChangeHandler = (lastCall) => { this.setState(() => ({ lastCall })) };

    nextCallChangeHandler = (nextCall) => { this.setState(() => ({ nextCall })) };

    nextHandler = () => { this.setState(() => ({ showFierstPart: false })); }

    backHandler = () => { this.setState(() => ({ showFierstPart: true })); }

    checkOfferHandler = () => {
        this.setState(() => ({
            showAddOfferModal: true,
            offerPostStatus: ''
        }));
    }

    closeHandlerModal = () => { this.setState(() => ({ showAddOfferModal: false })); }

    render () {
        return (
            <div className='content-container'>
                <h1 className='title__page'>Добави оферта</h1>
                <form>
                    {
                        this.state.showFierstPart ?
                            (<OfferPropsPartOne
                                values={this.state}
                                changeHandler={this.changeHandler}
                                nextHandler={this.nextHandler} />)
                            :
                            (<OfferPropsPartTwo
                                values={this.state}
                                changeHandler={this.changeHandler}
                                backHandler={this.backHandler}
                                prevCallChangeHandler={this.prevCallChangeHandler}
                                nextCallChangeHandler={this.nextCallChangeHandler}
                                checkOfferHandler={this.checkOfferHandler} />)
                    }
                </form>
                <EditOfferModal
                    isOpen={this.state.showAddOfferModal}
                    offerVlues={this.state}
                    closeHandlerModal={this.closeHandlerModal}
                    saveOfferHandler={this.saveOfferHandler}
                />

            </div>
        )
    }
}
const mapStateToProps = (state) => {
    return {
        offerPostStatus: state.offerPostStatus,
        editOffer: state.editOffer
    }
}
const actions = { postOffer, getCurrentOffer }

export default connect(mapStateToProps, actions)(EditOfferPage);
