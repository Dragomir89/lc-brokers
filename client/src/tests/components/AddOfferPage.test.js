import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import '../setupTests';
import { AddOfferPage } from '../../components/AddOfferPage';
import moment from 'moment';

test('should match snapshots correctly', () => {
    const wrapper = shallow(<AddOfferPage />);
    expect(wrapper.state('showFierstPart')).toBe(true);
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.find('#part-one').prop('nextHandler')();
    expect(wrapper.state('showFierstPart')).toBe(false);
    expect(toJSON(wrapper)).toMatchSnapshot();
});

test('should set new state correctly', () => {
    const wrapper = shallow(<AddOfferPage />);
    wrapper.find('#part-one').prop('changeHandler')({ target: { name: 'price', value: 100 } });
    expect(wrapper.state('price')).toBe(100);
});


test('OfferPropsPartTwo test props', () => {
    const wrapper = shallow(<AddOfferPage />);

    wrapper.find('#part-one').prop('nextHandler')();
    expect(wrapper.state('showFierstPart')).toBe(false);

    wrapper.find('#part-two').prop('prevCallChangeHandler')(moment(0));
    expect(wrapper.state('lastCall')).toEqual(moment(0));

    wrapper.find('#part-two').prop('nextCallChangeHandler')(moment(0));
    expect(wrapper.state('nextCall')).toEqual(moment(0));

    wrapper.find('#part-two').prop('checkOfferHandler')();
    expect(wrapper.state('showAddOfferModal')).toBe(true);
    expect(wrapper.state('offerPostStatus')).toBe('');

    wrapper.find('#part-two').prop('backHandler')();
    expect(wrapper.state('showFierstPart')).toBe(true);

});