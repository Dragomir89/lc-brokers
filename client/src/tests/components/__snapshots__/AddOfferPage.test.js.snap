// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots correctly 1`] = `
<div
  className="content-container"
>
  <h1
    className="title__page"
  >
    Добави оферта
  </h1>
  <form>
    <Connect(OfferPropsPartOne)
      changeHandler={[Function]}
      id="part-one"
      nextHandler={[Function]}
      values={
        Object {
          "address": "",
          "area": "",
          "constructionType": "",
          "constructionTypeId": "",
          "description": "",
          "floor": "",
          "info": "",
          "lastCall": "1970-01-01T00:00:00.000Z",
          "neighborhood": "",
          "neighborhoodId": "",
          "nextCall": "1970-01-01T00:00:00.000Z",
          "number": "",
          "offerPostStatus": "",
          "phoneNumber": "",
          "phoneNumber2": "",
          "phoneNumber3": "",
          "price": "",
          "propertyOwnerName": "",
          "propertyType": "",
          "propertyTypeId": "",
          "showAddOfferModal": false,
          "showFierstPart": true,
          "state": "",
          "stateVal": "",
        }
      }
    />
  </form>
  <Connect(AddOfferModal)
    closeHandlerModal={[Function]}
    isOpen={false}
    offerVlues={
      Object {
        "address": "",
        "area": "",
        "constructionType": "",
        "constructionTypeId": "",
        "description": "",
        "floor": "",
        "info": "",
        "lastCall": "1970-01-01T00:00:00.000Z",
        "neighborhood": "",
        "neighborhoodId": "",
        "nextCall": "1970-01-01T00:00:00.000Z",
        "number": "",
        "offerPostStatus": "",
        "phoneNumber": "",
        "phoneNumber2": "",
        "phoneNumber3": "",
        "price": "",
        "propertyOwnerName": "",
        "propertyType": "",
        "propertyTypeId": "",
        "showAddOfferModal": false,
        "showFierstPart": true,
        "state": "",
        "stateVal": "",
      }
    }
  />
</div>
`;

exports[`should match snapshots correctly 2`] = `
<div
  className="content-container"
>
  <h1
    className="title__page"
  >
    Добави оферта
  </h1>
  <form>
    <OfferPropsPartTwo
      backHandler={[Function]}
      changeHandler={[Function]}
      checkOfferHandler={[Function]}
      id="part-two"
      nextCallChangeHandler={[Function]}
      prevCallChangeHandler={[Function]}
      values={
        Object {
          "address": "",
          "area": "",
          "constructionType": "",
          "constructionTypeId": "",
          "description": "",
          "floor": "",
          "info": "",
          "lastCall": "1970-01-01T00:00:00.000Z",
          "neighborhood": "",
          "neighborhoodId": "",
          "nextCall": "1970-01-01T00:00:00.000Z",
          "number": "",
          "offerPostStatus": "",
          "phoneNumber": "",
          "phoneNumber2": "",
          "phoneNumber3": "",
          "price": "",
          "propertyOwnerName": "",
          "propertyType": "",
          "propertyTypeId": "",
          "showAddOfferModal": false,
          "showFierstPart": false,
          "state": "",
          "stateVal": "",
        }
      }
    />
  </form>
  <Connect(AddOfferModal)
    closeHandlerModal={[Function]}
    isOpen={false}
    offerVlues={
      Object {
        "address": "",
        "area": "",
        "constructionType": "",
        "constructionTypeId": "",
        "description": "",
        "floor": "",
        "info": "",
        "lastCall": "1970-01-01T00:00:00.000Z",
        "neighborhood": "",
        "neighborhoodId": "",
        "nextCall": "1970-01-01T00:00:00.000Z",
        "number": "",
        "offerPostStatus": "",
        "phoneNumber": "",
        "phoneNumber2": "",
        "phoneNumber3": "",
        "price": "",
        "propertyOwnerName": "",
        "propertyType": "",
        "propertyTypeId": "",
        "showAddOfferModal": false,
        "showFierstPart": false,
        "state": "",
        "stateVal": "",
      }
    }
  />
</div>
`;
